src/Microsoft.VisualStudio.LanguageServices.Razor/ForegroundThreadAffinitizedObject.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public class ForegroundThreadAffinitizedObject
```

Comment:
I think this is OK for now. I have some ideas about improving this, but I don't want to distract